[DOC] Adapt API/ABI stability documentation. (#67)
seqan/sharg-parser: doc/about/api/index.md

# Platform stability {#platform_stability}

The main requirement for Sharg is that your operating system provides one of the compilers supported by us.
In general, we only support the latest three major compiler versions.
We currently support the following compilers on 64-bit operating systems with little-endian CPU architectures:
  * GCC9, GCC10, GCC11

\note Only the most recent minor release of a compiler is guaranteed to be supported, e.g. when `gcc-10.4` is released,
we may drop support for `gcc-10.3`. Since all platforms with an older version receive minor release updates,
this should not be a problem.

We promise to support the above compilers in the latest release of Sharg, or until all the following
operating systems provide a newer supported compiler:

| Operating System             | Supported Releases¹                    |
|------------------------------|----------------------------------------|
| RedHat Enterprise Linux      | the latest release ²                   |
| CentOS Linux                 | the latest release ²                   |
| SUSE Linux Enterprise Server | the latest release                     |

> | Debian GNU/Linux             | "stable"                               |
> | Ubuntu Linux                 | the latest LTS release                 |

Author: ```suggestion
| Debian GNU/Linux             | "stable" and "old-stable"              |
| Ubuntu Linux                 | the two latest LTS releases            |
```

[TOC]

# API stability {#api_stability}

Sharg adheres to [semantic versioning](https://semver.org) and provides a stable API within

> one major release (all of `sharg-1.*.*`) unless otherwise noted.

Reviewer: ```suggestion
one major release unless otherwise noted.
```

I find it somehow confusing :D

> For details regarding API stability please see the
> [SeqAn part on API stability](http://docs.seqan.de/seqan/3-master-user/about_api.html#api_stability).

Reviewer: ```suggestion
For details regarding API stability, please see the
[SeqAn documentation on API stability](http://docs.seqan.de/seqan/3-master-user/about_api.html#api_stability).
```

# ABI stability

There is no ABI stability.

> For details regarding API stability please see the
> [SeqAn part on ABI stability](http://docs.seqan.de/seqan/3-master-user/about_api.html#autotoc_md38).

Reviewer: ```suggestion
For details regarding API stability, please see the
[SeqAn documentation on ABI stability](http://docs.seqan.de/seqan/3-master-user/about_api.html#autotoc_md38).
```

# Dependencies

> We will try to keep the Sharg Parser independent of any external libraries.

Reviewer: ```suggestion
We will keep Sharg independent of any external libraries, unless absolutely necessary.
```
